### PR TITLE
Update information on GAUCHE

### PR DESCRIPTION
**Pull request by jduerholt**
This PR adds correct source and liscense info for the molecular kernels based on gauche.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,6 +122,36 @@ We would love for you to use BoFire in your work! If you do, please cite [our pa
...
+        editor = {H. Larochelle and M. Ranzato and R. Hadsell and M.F. Balcan and H. Lin},
+        pages = {21524--21538},
+        publisher = {Curran Associates, Inc.},
+        title = {BoTorch: A Framework for Efficient Monte-Carlo Bayesian Optimization},
+        url = {https://proceedings.neurips.cc/paper_files/paper/2020/file/f5b1b89d98b7286673128a5fb112cb9a-Paper.pdf},
+        volume = {33},
+        year = {2020}
+    }
+
+For molecular optimizations, BoFire uses the molecular kernels from the [Gauche library](https://github.com/leojklarner/gauche). If you the molecular kernels in BoFire please cite also the [gauche paper](https://proceedings.neurips.cc/paper_files/paper/2023/hash/f2b1b2e974fa5ea622dd87f22815f423-Abstract-Conference.html):
```
Comment (Osburg) [inline README.md:139]:
```suggestion
For molecular optimizations, BoFire uses the molecular kernels from the [Gauche library](https://github.com/leojklarner/gauche). If you use the molecular kernels in BoFire please cite also the [gauche paper](https://proceedings.neurips.cc/paper_files/paper/2023/hash/f2b1b2e974fa5ea622dd87f22815f423-Abstract-Conference.html):
```

Comment (bertiqwerty) [inline README.md:139]:
```suggestion
For molecular optimizations, BoFire uses the molecular kernels from the [Gauche library](https://github.com/leojklarner/gauche). If you use the molecular kernels in BoFire please cite also the [gauche paper](https://proceedings.neurips.cc/paper_files/paper/2023/hash/f2b1b2e974fa5ea622dd87f22815f423-Abstract-Conference.html):
```

**Review by bertiqwerty — APPROVED** [review]
Thanks Johannes. Looks good. I found just one typo. See review-suggestion.

**Comment by jduerholt** [comment]
Thanks @Osburg and @bertiqwerty!

**Pull request merged**